tools/VscPreviewExtension/src/Proxy/requestArray.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { ProxyRequest } from "./proxyRequest";
+
+export class RequestArray {
+    private requests: { [id: string]: ProxyRequest } = {};
+    private _keys: string[] = [];
+
+    public add(request: ProxyRequest) {
+        if (this.exist(request)) {
+            this.update(request);
+        } else {
+            this.push(request);
+        }
+    }
+
+    public pop() {
```

Comment:
>pop [](start = 11, length = 3)

can shift method work for pop purpose? [http://codetunnel.com/9-javascript-tips-you-may-not-know/#queue](http://codetunnel.com/9-javascript-tips-you-may-not-know/#queue)